src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -580,7 +580,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_multx_sender_nic_index(MPIR_Comm * comm,
 {
     int nic_idx = 0;
 
-    if (MPIDI_OFI_COMM(comm).pref_nic) {
+    if (!comm->vcis_enabled) {
+        /* address exchange may not have performed on this comm */
+        nic_idx = 0;
+    } else if (MPIDI_OFI_COMM(comm).pref_nic) {
```

Comment:
Q for clarification: can a process still use multiple nics when a comm is using a single vci?

Reply:
Currently we call `MPIDI_Comm_set_vcis` in `MPID_Comm_commit_post_hook` for `MPI_COMM_WORLD`, so it is always enabled for comm world. The future plan is to make `comm_set_vcis` optional -- so users need explicitly set vcis in order to activate multiple nics -- calling `com_set_vcis` with a single vci is allowed, so multiple nics can be used with a single vci.

The design is to add some flexibility so that user or in the future we can enable vcis (and multiple nics) in a smaller comm than comm world and also when user need it.